File: Power Query/PQ_Challenge_156.R
library(tidyverse)
library(readxl)

input1 = read_excel("Power Query/PQ_Challenge_156.xlsx", range = "A1:B10")
input2 = read_excel("Power Query/PQ_Challenge_156.xlsx", range = "D1:E5")

test = read_excel("Power Query/PQ_Challenge_156.xlsx", range = "G1:K6")

result = input1 %>% 
  left_join(input2, by = "Subjects") %>%
  pivot_wider(names_from = "Subjects", values_from = "Teacher", values_fill =  NA_character_) %>%
  select(Name, Biology, Chemistry, Geology, Physics)

identical(result, test)
# [1] TRUE

File: Power Query/100-199/156/PQ_Challenge_156.py
import pandas as pd

input1 = pd.read_excel("PQ_Challenge_156.xlsx", usecols="A:B", nrows=10)
input2 = pd.read_excel("PQ_Challenge_156.xlsx", usecols="D:E", nrows=5)
test = pd.read_excel("PQ_Challenge_156.xlsx", usecols="G:K", nrows=6)

result = input1.merge(input2, on="Subjects", how="left")
result = result.pivot(index="Name", columns="Subjects", values="Teacher").reset_index()
result = result[["Name", "Biology", "Chemistry", "Geology", "Physics"]]

print(result.equals(test))
